cranelift/codegen/src/ir/function.rs

```diff
@@ -96,6 +97,18 @@ pub struct Function {
     /// are saved in the frame. This information is created during the prologue and epilogue
     /// passes.
     pub frame_layout: Option<FrameLayout>,
+
+    /// An optional closure which calculates the stack limit for this function
+    /// from the arguments of the function.
+    ///
+    /// When configured a stack check will be emitted in the prologue of this
+    /// function, trapping if the stack check fails. This closure, if specified,
+    /// can be then used to infer the stack limit from provided arguments as an
+    /// alternative to using `ArgumentPurpose::StackLimit`.
+    ///
+    /// The first argument is the cursor we're encoding the prologue too, and
+    /// the second argument is a scratch register location if necessary.
+    pub stack_limit_from_arguments: Option<fn(&mut EncCursor, ir::ValueLoc) -> ir::Value>,
```

Comment:
How is this serialized to a .clif file?

Reply:
It is not.

Reply:
Maybe use a `GlobalValue` here instead. That way it is serializable.

Reply:
I'm not sure if that would work out because of how manual this is due to how late in the pipeline this executes. I'm also pretty unfamiliar with cranelift itself. If you've got an example of how to do this that would be great!

Reply:
If you can use a `GlobalValueData::Load` to load from a `GlobalValueData::VMContext`. Then you set this field to the `GlobalValue` of the `GlobalValueData`. You can then during the stack check codegen use `global_value` to get the stack limit

Reply:
I [tried that](https://gist.github.com/alexcrichton/c43e66cc13154a97be2af665dc1aa117) but [got a panic](https://gist.github.com/alexcrichton/02697129b5a92ad2b8983a74115b244a) that I think means the legalizer needs to run, but this is happening after the legalizer has already run.

Reply:
It seems that `EncCursor` doesn't allow inserting instructions that need to be legalized.